### DISCUSSION TITLE
[@types/redis] Missing `user` in `ClientOpts`

### DISCUSSION THREAD

**Discussion by khangly** [DefinitelyTyped/DefinitelyTyped#57338] [Issues with a @types package] [answered]
Redis version 6 added Access Control List (ACL), and the [client](https://www.npmjs.com/package/redis) was also updated to have the `user` as an argument in `createClient`. Please update the type to have `user` in `ClientOpts` as well.

[a comment by an automated account is not shown]

**Comment by 43081j** [comment, marked as answer]
I feel like `@types/redis` should be deleted if anything.

node-redis now ships its own types, you shouldn't need these.

EDIT: ah i see, the stable one on npm is still 3.x of the client, which does _not_ have types. but when 4.x lands, it will.

you could use `redis@next` for now (`npm i redis@next`) to get 4.x which comes with types.

**Reply by khangly** [reply]
Yup, version 4 has support for typescript and was just released a day after you commented!
